src/Symfony/Bundle/FrameworkBundle/Tests/Functional/SerializerTest.php

```diff
@@ -52,20 +53,29 @@ public function testNormalizersAndEncodersUseDefaultContextConfigOption(string $
         self::assertEquals('foo', $defaultContext['fake_context_option']);
     }
 
-    public static function provideNormalizersAndEncodersWithDefaultContextOption(): array
+    public static function provideNormalizersAndEncodersWithDefaultContextOption(): iterable
     {
-        return [
-            ['serializer.normalizer.constraint_violation_list.alias'],
-            ['serializer.normalizer.dateinterval.alias'],
-            ['serializer.normalizer.datetime.alias'],
-            ['serializer.normalizer.json_serializable.alias'],
-            ['serializer.normalizer.problem.alias'],
-            ['serializer.normalizer.uid.alias'],
-            ['serializer.normalizer.object.alias'],
-            ['serializer.encoder.xml.alias'],
-            ['serializer.encoder.yaml.alias'],
-            ['serializer.encoder.csv.alias'],
-        ];
+        $kernel = static::createKernel(['test_case' => 'Serializer', 'root_config' => 'default_context.yaml']);
+        $container = $kernel->getContainerWithLoadedExtensions();
+
+        $services = array_merge(
+            $container->findTaggedServiceIds('serializer.normalizer'),
+            $container->findTaggedServiceIds('serializer.encoder')
+        );
+        foreach ($services as $serviceId => $attributes) {
+            $class = $container->getDefinition($serviceId)->getClass();
+            if (null === $reflectionConstructor = (new \ReflectionClass($class))->getConstructor()) {
+                continue;
+            }
+            foreach ($reflectionConstructor->getParameters() as $reflectionParam) {
+                if ('array $defaultContext' === $reflectionParam->getType().' $'.$reflectionParam->getName()) {
+                    yield [$serviceId.'.alias'];
+                    break;
+                }
+            }
+        }
```

Comment:
I made this dynamic in case new normalizers/encoders were added. Is it worth it, no idea.